frontend/src/app/modules/members/invite/invite.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { BaseComponent } from '@core/components/base/base.component';
+import { MemberService } from '@core/services/member.service';
+import { Member } from '@shared/models/member/member';
+import { Subject } from 'rxjs';
+import { debounceTime, switchMap } from 'rxjs/operators';
+import { RoleService } from '@core/services/role.service';
+import { Role } from '@shared/models/role/role';
+import { TeamOption } from '@shared/models/teams/team-option';
+import { NewMember } from '@shared/models/member/new-member';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthenticationService } from '@core/services/authentication.service';
+import { User } from '@shared/models/user/user';
+import { Organization } from '@shared/models/organization/organization';
+import { ToastNotificationService } from '@core/services/toast-notification.service';
+import { TeamService } from '@core/services/team.service';
+import { UserService } from '@core/services/user.service';
+import { ShareDataService } from '@core/services/share-data.service';
+import { Invition } from '@shared/models/member/invition';
+
+@Component({
+    selector: 'app-invite',
+    templateUrl: './invite.component.html',
+    styleUrls: ['./invite.component.sass']
+})
+export class InviteComponent extends BaseComponent implements OnInit {
+    @Output() closeModal = new EventEmitter<void>();
+    notMembers: User[];
+    searchTerm: Subject<string> = new Subject<string>();
+    roles: Role[];
+    teams: TeamOption[];
+
+    loadingNumber: number = 0;
+
+    user: User;
+    organization: Organization;
+
+    invations: Invition[] = [{ member: {} as NewMember, groupForm: this.generateGroupForm() }];
+    constructor(
+        private memberService: MemberService,
+        private roleService: RoleService,
+        private authService: AuthenticationService,
+        private toastNotifications: ToastNotificationService,
+        private teamService: TeamService,
+        private userService: UserService,
+        private updateDataService: ShareDataService<Member>
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.loadData();
+        this.user = this.authService.getUser();
+    }
+
+    generateGroupForm() {
+        return new FormGroup({
+            name: new FormControl(
+                '',
+                [
+                    Validators.required,
+                ]
+            ),
+            team: new FormControl(
+                []
+            ),
+            role: new FormControl(
+                '',
+                [
+                    Validators.required,
+                ]
+            )
+        });
+    }
+
+    loadData() {
+        this.searchTerm.pipe(
+            this.untilThis,
+            debounceTime(300),
+            switchMap((term: string) => {
+                this.loadingNumber += 1;
```

Comment:
What is it loading number?

Reply:
If it bigger than 0, then loading else no) It's done to prevent stop loading when some of requests are completed but not all

Reply:
It looks like work around more than solution of work